### PR TITLE
Feat/case study/create

### PR DESCRIPTION
**Pull request by Cosmoboi-i**

### CODE REVIEW

File: src/controllers/case-study.controller.js [hunk 1]
```diff
@@ -1,8 +1,20 @@
 const caseStudyServices = require('../services/case-study.service');
+
+const createCaseStudy = async (req, res) => {
+  try {
+    const caseStudy = await caseStudyServices.createCaseStudy(req.body);
```
Comment (rohini-senthil) [inline src/controllers/case-study.controller.js:5]: Can we add logging like how it is done for rest of the endpoints?

Comment (Cosmoboi-i) [reply src/controllers/case-study.controller.js:5]: Sure, on it

File: src/services/case-study.service.js [hunk 2]
```diff
@@ -1,6 +1,37 @@
...
 const db = require('../models');
 const logger = require('../logger');
 
+const updateUserFk = async (collaboratorsIds, caseStudyId) => {
+  for (let collabId of collaboratorsIds) {
+    user = userService.getUser(collabId);
+    user.dataValues['case_studies_ids'].push(caseStudyId);
+    userService.updateUser(collabId, user);
+  }
+};
```
Comment (rohini-senthil) [inline src/services/case-study.service.js:12]: Can directly call update user service function instead of this new method

File: src/services/case-study.service.js [hunk 3]
```diff
@@ -1,6 +1,37 @@
...
+
+const updateEngagementFk = async (engagementId, caseStudyId) => {
```
Comment (rohini-senthil) [inline src/services/case-study.service.js:14]: Can directly call update engagement service function instead of this new method

File: src/services/case-study.service.js [hunk 4]
```diff
@@ -1,6 +1,37 @@
...
+
+const createCaseStudy = async caseStudy => {
+  const newCaseStudy = await case_studies.create(caseStudy);
+  const { caseStudyId, collaboratorsIds, engagementId } = newCaseStudy.dataValues;
+
+  // update users case_study_ids
+  updateUserFk(collaboratorsIds, caseStudyId);
+
+  // update engagements case_study_ids
+  updateEngagementFk(engagementId, caseStudyId);
```
Comment (rohini-senthil) [inline src/services/case-study.service.js:30]: Let's move this one level above to the controller as we will be calling different service (user/engagement) methods

Comment (Cosmoboi-i) [reply src/services/case-study.service.js:33]: The updateFk function will take more than one line, so I can either remove these functions and put them in the service, or keep these functions and include the one liner function in the controller

Comment (Cosmoboi-i) [reply src/services/case-study.service.js]: I have gone with the former way, please have a look

Comment (rohini-senthil) [reply src/services/case-study.service.js]: It should either way be in controller since they are different entities – let's move them to the controller

File: database/config/config.json [hunk 5]
```diff
@@ -2,17 +2,26 @@
   "development": {
```
Comment (rohini-senthil) [inline database/config/config.json:2]: Let's revert the changes on this file

File: src/controllers/case-study.controller.js [hunk 6]
```diff
@@ -1,10 +1,41 @@
...
+      console.log(user);
+      user['caseStudyIds'].push(caseStudyId);
+      userServices.updateUser(collabId, user);
+    }
+
+    // update engagements case_study_ids
+    const engagement = await engagements.findByPk(engagementId);
+    console.log(engagement);
+    engagement['caseStudyIds'].push(caseStudyId);
+    engagement.save();
```
Comment (rohini-senthil) [inline src/controllers/case-study.controller.js:25]: Can we make this into a service function?

File: src/services/case-study.service.js [hunk 7]
```diff
@@ -1,6 +1,14 @@
+const { case_studies } = require('../models');
+const userService = require('./user.service');
```
Comment (rohini-senthil) [inline src/services/case-study.service.js:2]: This import can be removed since it is not used now

**Review by rohini-senthil — APPROVED** [review]
